master/buildbot/process/properties.py

```diff
@@ -743,3 +743,27 @@ def getRenderingFor(self, build):
         return d
 
 registerAdapter(_DictRenderer, dict, IRenderable)
+
+
+class Transform(object):
+
+    """
+    A renderable that combines other renderables' results using an arbitrary function.
+    """
+
+    implements(IRenderable)
+
+    def __init__(self, function, *args, **kwargs):
+        if not callable(function) and not IRenderable.providedBy(function):
+            config.error("function given to Transform neither callable nor renderable")
+
+        self.function = function
```

Comment:
I know that we do not consistently follow this, but for new code (at least) I prefer those members to be "private":

```python
    self._function = function
    ...
```

In the past we had quite a number of nightmares related to API management (since all non-private data members are essentially part of the API).